src/azure-cli/azure/cli/command_modules/backup/custom_help.py

```diff
@@ -671,11 +672,25 @@ def validate_and_extract_container_type(container_name, backup_management_type):
     return None
 
 
-def validate_update_policy_request(existing_policy, new_policy):
+def validate_update_policy_request(existing_policy, new_policy, yes=False):
     existing_backup_management_type = existing_policy.properties.backup_management_type
     new_backup_management_type = new_policy.properties.backup_management_type
     if existing_backup_management_type != new_backup_management_type:
         raise CLIError("BackupManagementType cannot be different than the existing type.")
+    # vault -> snapshot
+    if hasattr(existing_policy.properties, 'vault_retention_policy') and existing_policy.properties.vault_retention_policy is not None and hasattr(new_policy.properties, 'retention_policy') and new_policy.properties.retention_policy is not None:
+        raise CLIError(
+            """
+            Switching the backup tier from vaulted backup to snapshot is not possible. 
+            Please create a new policy for snapshot-only backups.
+            """)
+    # snapshot -> vault
+    if not yes and hasattr(existing_policy.properties, 'retention_policy') and existing_policy.properties.retention_policy is not None and hasattr(new_policy.properties, 'vault_retention_policy') and new_policy.properties.vault_retention_policy is not None:
```

Comment:
If `yes` is passed as `True`, we never enter this path anyway. If the goal is to have `yes` skip the warning prompt, we need to remove the check from this line and add it lower down in 692: `if yes or not prompt_y_n(warning_prompt)`.